core/commands/files.go

```diff
@@ -757,6 +759,22 @@ stat' on the file or any of its ancestors.
 			return
 		}
 
+		dirtomake := gopath.Dir(path)
+
+		if dirtomake != "/" && dashp == true {
+			root := nd.FilesRoot
+
+			err = mfs.Mkdir(root, dirtomake, mfs.MkdirOpts{
```

Comment:
Given that `getFileHandle` already has special logic for creating missing files, I'd prefer to add this logic to `getFileHandle`, replacing the the `create` bool with a [constant](https://splice.com/blog/iota-elegant-constants-golang/). However, I don't feel too strongly about this.